### ISSUE TITLE
Feedback: UI :speech_balloon:

### ISSUE THREAD

**Issue by anbraten** [woodpecker-ci/woodpecker#468]
Summary of UI issues and smaller changes (based on new UI from #245) which can be posted as comments. For complex improvements please open a separate issue.

# UI improvements
- [x] improve responsive (mobile) usage
- [x] improve darkmode favicon #576 
- [x] favicon to show pipeline status #576 
- [x] use textarea for secret values (to support long texts like ssh keys) #574 
- [x] show PR name / id instead of target branch if `ref: refs/pull/xxx/merge`
- [x] Show matrix options in build view #526 
- [x] add page `/<user-or-group-name>` to view all enabled woodpecker repos of a user / group
- [x] use icons for build proc status (currently just colored points)

# UI bugs
- [x] #469 
- [x] fix overflowing commit messages #528
- [x] fix back button (direct page load) to go back in history or to some default if history is empty #529 
- [x] fix new build updates not being used / send (new builds are not added list)
- [x] fix flickering when adding new log-lines while step is running

Next Round:  #1314

**Comment by 6543** [comment]
screenshot in README.md need an update ;)

**Comment by HerHde** [comment]
`// The new UI looks great, thanks!`
Display the version number or hash somewhere (like `next-f00baa` for examle before)

**Comment by HerHde** [comment]
The sidebar with the jobs on the right becomes scrollable for long words in commit messages.

![image](https://user-images.githubusercontent.com/7630662/141981066-bdfd92c7-2db1-48ec-a26b-ada7546bf2c3.png)

**Comment by HerHde** [comment]
Change secret values from input fields to textareas again (or change it dynamically) to allow multi-line values like SSH-Keys again.

**Comment by HerHde** [comment]
From a repo's builds (and also settings), provide a link back to the repository itself. Preferably with the name of the repo itself. Maybe as a breadcrumb navigation? The arrow back button only goes back in history.

**Comment by catdevnull** [comment]
https://github.com/woodpecker-ci/woodpecker/pull/526 fixes Show matrix options in build view

**Comment by anbraten** [comment]
> `// The new UI looks great, thanks!` Display the version number or hash somewhere (like `next-f00baa` for examle before)

I still have that below the logo. Which version do you use? Could you attach a screenshot pls.

**Comment by catdevnull** [comment]
> I still have that below the logo. Which version do you use? Could you attach a screenshot pls.

Not who you answered to, but here's my Woodpecker on the `:latest` container tag:

![image](https://user-images.githubusercontent.com/86896666/142957200-e4c6fa56-7a37-4f8e-8c73-ee480c531033.png)

**Comment by HerHde** [comment]
In a project's settings under `Registires`, the info button links to a [404 in the docs](https://woodpecker-ci.org/docs/usage/registry).

![image](https://user-images.githubusercontent.com/7630662/143479001-5a05909b-f6ff-4fee-bc38-a62c7010d16a.png)

**Comment by 6543** [comment]
https://codeberg.org/Codeberg-CI/feedback/issues/31#issuecomment-284005

.. repo settings -> secret -> if pulls are checked show warning

**Comment by 6543** [comment]
different icon for different links

this should show a pull icon instead of a commit:
![.](https://user-images.githubusercontent.com/24977596/144715734-2ded1bba-e3fa-4d88-be84-fa055e18174a.png)

**Comment by 6543** [comment]
https://github.com/woodpecker-ci/woodpecker/pull/593#issuecomment-991585391
- add a label/tag `disabled` on /repos view for disabled repos

**Comment by 6543** [comment]
https://github.com/woodpecker-ci/woodpecker/pull/593#issuecomment-991585391
- on disabled repos -> repo config -> repo actions: deactivate button should be hidden
- on disabled repos -> repo config -> repo actions: add activate button

**Comment by 6543** [comment]
for live log view: "auto follow" option

**Comment by anbraten** [comment]
show proc errors

**Comment by anbraten** [comment]
![Screenshot from 2022-01-15 17-38-00](https://user-images.githubusercontent.com/6918444/149629902-bbbfb1b9-27ee-479d-81dc-8cee2ed6c877.png)

Link tags to SCM system. Hide `refs/tags/` from tag name.

**Comment by 6543** [comment]
horizontal scroll-bar for  config view:
![Bildschirmfoto zu 2022-01-16 18-40-38](https://user-images.githubusercontent.com/24977596/149671381-54bc120a-c969-42cc-a01d-85458ee81724.png)

**Comment by eladyn** [comment]
Couldn't find something similar in the repository, feel free to hide if this was already addressed:

![woodpecker_ui_bug](https://user-images.githubusercontent.com/59307989/151639310-94f87ef8-19a0-4165-acdb-4b1b71b39f08.png)

In WebKit (Safari, GNOME Web, ...) the animated circle indicator is set off from the correct location, doesn't happen in Chromium or Firefox.

**Comment by 6543** [comment]
if you are in buildView and login, go back to this buildView again

**Comment by valentinleistner** [comment]
It would be nice if you could see the current build status and maybe last commit and branch on the repos page

<img width="1322" alt="screenshot of the current repos page" src="https://user-images.githubusercontent.com/77448526/152203360-3007bf8e-40c4-4287-8680-9f9222e7a211.png">

**Comment by anbraten** [comment]
Pipelines in branches tab (for example master) also include Pull-requests against master => should not contain PRs

**Comment by qwerty287** [comment]
Why are the user overview (#741) and the branches list only available if I'm logged in, even if the repo is public? Is there any reason for this?

**Comment by anbraten** [comment]
I think it is simply missing a parameter in the Vue router config which enforces authentication for the page. I never checked if #741 allows you to see the projects of other users, we have to check that as well.

**Comment by qwerty287** [comment]
I checked it, and you can't access them, this makes it understandable why you have to be logged in. However, I'd say it would be nice to see public projects of other users.

**Comment by 6543** [comment]
multi pipeline: make them able to "colaps" (to hide/show steps, similar to github pull diff view ...)

**Comment by mrcates** [comment]
I’d love to be able to run manual/ad-hoc pipelines from the UI in addition to the wonderful automatic triggers that exist. I would use this feature either for running sysadmin tasks (ansible, bash, etc.)

**Comment by ivictbor** [comment]
I believe WCAG compatibility is pretty important, at least AA grade. A lot of proffesianal interfaces developers (web/mobile) might use this CI, it is great! But a lot of us know WCAG requirments wery well (current issues in Woodpecker are notacible by human eye), so when we see Woodpecker's "gray on gray" (see this issue https://github.com/woodpecker-ci/woodpecker/issues/721 ) it is kinda looks like, you know, ummm, a little bit "rookie designer mistake". This is absolutely OK, we all passed these steps, just sharing info how to improve and look more professional.
BTW drone satisfied this requirement (you might notice contrast white on blue an so on), you selected very cool color scheme too, only left to fix contrast of texts a little bit.

**Comment by qwerty287** [comment]
Since [my comment](https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1046300145) has been hidden while my idea isn't fixed, I'll just comment again:
I think it would be nice to see public projects of other users on their "profile" list.

**Comment by silverwind** [comment]
I'd like to see this "Avenir" custom font removed and instead a standard [system font stack](https://systemfontstack.com/) to be used. Benefits are a consistent experience across multiple apps that also use system font stack, as well as faster load times and no flash of wrong fonts.

See [here](https://github.com/go-gitea/gitea/blob/ba0a0d31c97a1cfc4f163bdd32b1749c4ddc74a5/web_src/less/_base.less#L3-L5) for an example.

**Comment by tecosaur** [comment]
This seems like the place to plop UI feedback? I've just started using Woodpecker and I'm sorry to say that the dark mode contrast is atrocious.

![image](https://user-images.githubusercontent.com/20903656/183231941-f30595e9-0604-4ca8-9530-395830021366.png)

Let alone WCAG AAA, this fails WCAG AA standards of accessibility.

![image](https://user-images.githubusercontent.com/20903656/183232110-aab14b51-f757-4176-947c-9686d80e4e8b.png)
![image](https://user-images.githubusercontent.com/20903656/183232124-3ef03ce3-b9b5-40ee-a78c-30a9da5e8d78.png)

I recommend https://accessible-colors.com/ for checking that there's sufficient contrast between foreground and background colours.

Here's what `#CCCFD4` looks like:

![image](https://user-images.githubusercontent.com/20903656/183232170-a6322f94-ce30-4110-9922-b87603b25f21.png)

On a more stylistic note, I don't think the dark borders do the design any favours either. Here's a lighter version:

![image](https://user-images.githubusercontent.com/20903656/183232207-87b3e3ab-bac7-42ec-9ca2-3777f825b268.png)

There are other improvements that can be made, but this would be a good starting point I think :slightly_smiling_face:.

**Comment by tecosaur** [comment]
Actually, on this topic, how would the project feel about taking heavy inspiration from Gitea's default theme?

**Comment by anbraten** [comment]
@tecosaur Did you checked next version?

**Comment by tecosaur** [comment]
I'm currently running 0.15.3, I might try master in the near future.

**Comment by ivictbor** [comment]
@tecosaur please check https://ci.woodpecker-ci.org/woodpecker-ci/woodpecker, I did not check with contrast calculator, but by human eye check it looks much better!

**Comment by tecosaur** [comment]
Thanks for that link, I wasn't aware of that! Good to see that it's been improved, though I do think it's only partway there.

For example, here's what I'm currently seeing:

![image](https://user-images.githubusercontent.com/20903656/183242944-bd625370-4efa-4b67-9c11-802e4dcf48b1.png)

If I just copy some styling from Gitea's default theme:

![image](https://user-images.githubusercontent.com/20903656/183242958-f0559131-778b-4ae2-9ca0-ebd3cb48e6b2.png)

I find the latter markedly better.

**Comment by ivictbor** [comment]
> Thanks for that link, I wasn't aware of that! Good to see that it's been improved, though I do think it's only partway there.
> 
> For example, here's what I'm currently seeing:
> 
> ![image](https://user-images.githubusercontent.com/20903656/183242944-bd625370-4efa-4b67-9c11-802e4dcf48b1.png)
> 
> If I just copy some styling from Gitea's default theme:
> 
> ![image](https://user-images.githubusercontent.com/20903656/183242958-f0559131-778b-4ae2-9ca0-ebd3cb48e6b2.png)
> 
> I find the latter markedly easier to read.

Agree about minimalizm of colors in rows
Also clickable elements should have a separate color (like green repo link), otherwise user would not even know that they are clickable

**Comment by tecosaur** [comment]
Woodpecker is already roughly *similar* to Gitea in design (e.g. the background colour is an exact match). For ease of use and accessibility having a clear design is important, and simply put it's much easier to use an existing project as a reference than coming up with something good from scratch.

I know I've only just started looking at this project, but perhaps a good avenue for initially improving the design/styling would be to take heavy inspiration from Gitea's default theme? In the slightly longer term, it would also be neat for Woodpecker to accept custom themes as Gitea does. That would make it easier to gradually develop a distinct visual identity (should the project want that) in parallel to a well-tested style and then switch it as the default once it's reached a sufficient level of refinement.

**Comment by tecosaur** [comment]
Cool, I'll see about doing so then. One quick question, if I want to try playing around with the styling myself is there some main CSS file like https://github.com/go-gitea/gitea/blob/main/web_src/less/themes/theme-arc-green.less that I could try tweaking?

**Comment by 6543** [comment]
Create design files #1122

**Comment by 6543** [comment]
> Pipelines in branches tab (for example master) also include Pull-requests against master => should not contain PRs

I would say they should, you can target PRs to different branches

**Comment by dvjn** [comment]
Hi, I've recently started using woodpecker. Awesome tool, but imho bad ui (inconsistent spacing and design of ui elements, bad dark mode contrast, etc).

I'd really like to contribute some theme improvements with a PR.
Could someone please give me some pointers on how to setup up a dev environment for this project?

**Comment by 6543** [comment]
Next Round: https://github.com/woodpecker-ci/woodpecker/issues/1314

open comments will be copied ...

**Comment by 6543** [comment]
for contrast related stuff ( accessibility ) there is now  #1315
